### PR TITLE
Add a delay to laser profile read from "txye" file

### PR DESCRIPTION
**Pull request by lucafedeli88**
This PR allows delaying or anticipating a laser read from a "txye" file.

Note that, although `delay` has been added to CI test, its value is zero, so we are not (yet) testing thoroughly this feature.

### CODE REVIEW

File: Docs/source/running_cpp/parameters.rst [hunk 1]
```diff
@@ -816,11 +816,13 @@ Laser initialization
       function, they still have to be specified as they are used for numerical purposes.
     - ``"from_txye_file"``: the electric field of the laser is read from an external binary file
       whose format is explained below. It requires to provide the name of the binary file
-      setting the additional parameter ``<laser_name>.txye_file_name`` (string). It accepts an
-      optional parameter ``<laser_name>.time_chunk_size`` (int). This allows to read only
+      setting the additional parameter ``<laser_name>.txye_file_name`` (``string`). It accepts an
```
Comment (RemiLehe) [inline Docs/source/running_cpp/parameters.rst]:
```suggestion
      setting the additional parameter ``<laser_name>.txye_file_name`` (`string`). It accepts an
```

File: Docs/source/running_cpp/parameters.rst [hunk 2]
```diff
@@ -816,11 +816,13 @@ Laser initialization
...
+      optional parameter ``<laser_name>.time_chunk_size`` (`int`). This allows to read only
       time_chunk_size timesteps from the binary file. New timesteps are read as soon as they are needed.
       The default value is automatically set to the number of timesteps contained in the binary file
       (i.e. only one read is performed at the beginning of the simulation).
+      It also accepts the optional parameter ``<laser_name>.delay`` (`float`; in seconds), which allows
+      delaying (``delay`` > 0) or anticipating (``delay < 0``) the laser by the specified amount of time.
```
Comment (RemiLehe) [inline Docs/source/running_cpp/parameters.rst]:
```suggestion
      delaying (``delay > 0``) or anticipating (``delay < 0``) the laser by the specified amount of time.
```

**Review by RemiLehe — CHANGES_REQUESTED** [review]
Thanks for this PR! I have a two small typographic corrections.

**Comment by RemiLehe** [comment]
@lucafedeli88 There seems to be a merge conflict. Would you have time to resolve it?

**Comment by lucafedeli88** [comment]
> @lucafedeli88 There seems to be a merge conflict. Would you have time to resolve it?

It should be OK now...

**Review by RemiLehe — APPROVED** [review]

**Pull request merged**